tests/integration_tests/conftest.py

```diff
@@ -310,28 +327,38 @@ def _client(
             pytest.skip("lxd_setup requires LXD")
         local_launch_kwargs["lxd_setup"] = lxd_setup
 
-    with session_cloud.launch(
-        user_data=user_data, launch_kwargs=launch_kwargs, **local_launch_kwargs
-    ) as instance:
-        if lxd_use_exec is not None and isinstance(
-            instance.instance, LXDInstance
-        ):
-            # Existing instances are not affected by the launch kwargs, so
-            # ensure it here; we still need the launch kwarg so waiting works
-            instance.instance.execute_via_ssh = False
-        previous_failures = request.session.testsfailed
-        yield instance
-        test_failed = request.session.testsfailed - previous_failures > 0
-        _collect_artifacts(instance, request.node.nodeid, test_failed)
-    # conflicting requirements:
-    # - pytest thinks that it can cleanup loggers after tests run
-    # - pycloudlib thinks that at garbage collection is a good place to tear
-    #   down sftp connections
-    # After the final test runs, pytest might clean up loggers which will cause
-    # paramiko to barf when it logs that the connection is being closed.
-    #
-    # Manually run __del__() to prevent this teardown mess.
-    instance.instance.__del__()
+    try:
+        with session_cloud.launch(
+            user_data=user_data,
+            launch_kwargs=launch_kwargs,
+            **local_launch_kwargs,
+        ) as instance:
+            if lxd_use_exec is not None and isinstance(
+                instance.instance, LXDInstance
+            ):
+                # Existing instances are not affected by the launch kwargs, so
+                # ensure it here; we still need the launch kwarg so waiting
+                # works
+                instance.instance.execute_via_ssh = False
+            previous_failures = request.session.testsfailed
+            yield instance
+            test_failed = request.session.testsfailed - previous_failures > 0
+            _collect_artifacts(instance, request.node.nodeid, test_failed)
+        # conflicting requirements:
+        # - pytest thinks that it can cleanup loggers after tests run
+        # - pycloudlib thinks that at garbage collection is a good place to
+        # tear down sftp connections
+        #
+        # After the final test runs, pytest might clean up loggers which will
+        # cause paramiko to barf when it logs that the connection is being
+        # closed.
+        #
+        # Manually run __del__() to prevent this teardown mess.
+        instance.instance.__del__()
+    except Exception as e:
+        pytest.exit(
```

Comment:
I don't think we want this here. If a single instance launch fails it'll exit the whole suite.